Mage/src/main/java/mage/abilities/common/CaseAbility.java

```diff
@@ -85,7 +86,8 @@ public CaseAbility(Ability initialAbility, Condition toSolveCondition, Ability s
             if (solvedAbility instanceof SimpleStaticAbility) {
                 for (Effect effect : solvedAbility.getEffects()) {
                     if (!(effect instanceof ConditionalContinuousEffect ||
-                            effect instanceof ConditionalAsThoughEffect)) {
+                            effect instanceof ConditionalAsThoughEffect ||
+                            effect instanceof ConditionalReplacementEffect)) {
                         throw new IllegalArgumentException("solvedAbility must be one of ConditionalActivatedAbility, " +
```

Comment:
Use "Wrong code usage: your message" -- it's a standard code style for such use cases.